### DISCUSSION TITLE
Access Control through Subfolders

### DISCUSSION THREAD

**Discussion by crabnebul** [apache/airflow#24853] [General] [answered]
I am trying to set up Airflow for all users on my team without creating any interference between different users. However, Airflow permissions appear not granular enough to allow for full `read`, `delete`, and `edit` actions on subdirectories.

Am I missing something, is there a workaround, or is this feature simply unavailable?

```
$AIRFLOW_HOME/dags/

users1/
users2/
...
usern/
```

**Comment by potiuk** [comment, marked as answer]
It's unavailable. This is part of future multi-tenancy feature which is being implemented. but it will take quite some time to get there.

**Reply by crabnebul** [reply]
Let me follow up on my question one more time. 
How would you suggest managing multi-user access in absence of this functionality?

**Reply by potiuk** [reply]
Let me repeat the answer. There is no "good official way of doing it". Airflow is not (so far) capable of multi-tenancy. Full stop.

In the future it likely will be https://airflowsummit.org/sessions/2022/multitenancy-is-coming/ 

You can take a look at how others did it in a workaround fashion (for example https://airflowsummit.org/sessions/2021/dynamic-security-roles-in-airflow-for-multi-tenancy/) from 2021 summit (and you can look through ohter summit videos from 2020 and 2022 there were at least few talks about it). But those solutions have multiple flaws (you will find details in the video)

**Reply by potiuk** [reply]
BTW. The best way currently is to have separate Airflow instance for each team.

**Reply by crabnebul** [reply]
Awesome. Thanks for your answer.
